src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -893,13 +890,8 @@ private ReadOnlySpan<byte> UnescapeString(in DbRow row, out ArraySegment<byte> r
                 return text;
             }
 
-            int idx = text.IndexOf(JsonConstants.BackSlash);
-            Debug.Assert(idx >= 0);
-
             byte[] rent = ArrayPool<byte>.Shared.Rent(length);
-            text.Slice(0, idx).CopyTo(rent);
-
-            JsonReaderHelper.Unescape(text, rent, idx, out int written);
+            JsonReaderHelper.Unescape(text, rent, out int written);
             rented = new ArraySegment<byte>(rent, 0, written);
             return rented.AsSpan();
```

Comment:
```suggestion
            return rent.AsSpan(..written);
```

Reply:
`return rent.AsSpan(0, written);` is slightly better

Reply:
Yea and less pretty

Reply:
The compilers (Roslyn and / or JIT) should emit the same code for both cases ideally 😉 
Cf. https://github.com/dotnet/roslyn/issues/43598#issuecomment-876461779

Reply:
While I agree, the linked example is different (and easier).

Reply:
The `ArraySegment` is passed to the `out` parameter so that the rented buffer can be returned and cleared, see the `ClearAndReturn` method below:
https://github.com/dotnet/runtime/blob/8d4a724883715cc5ae71e020b22762136f7ce50b/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L907-L914

Returning both the ArraySegment _and_ the ROS is probably overdoing it, but I'll avoid making that refactoring in this PR.